server/src/main/java/org/candlepin/policy/js/compliance/hash/HashableStringGenerators.java

```diff
@@ -211,7 +211,15 @@ public String generate(Consumer target) {
             }
 
             String generated = target.getUuid();
-            generated += generateFromCollection(target.getFacts().entrySet(), STRING_ENTRY);
+
+            if (target.getFacts() != null) {
+                generated += generateFromCollection(target.getFacts().entrySet(), STRING_ENTRY);
+            }
+            else {
+                // Append null if consumer facts are null.
+                generated += "null";
```

Comment:
What is the purpose of adding this? Couldn't we just not append anything (skip the else alltogether)?

Reply:
In [generateFromCollection()](https://github.com/candlepin/candlepin/blob/master/server/src/main/java/org/candlepin/policy/js/compliance/hash/HashableStringGenerators.java#L66-L69) method we append null if any collection is null. I skipped the method call & did it over here.

Reply:
Perhaps it would be better to reduce the confusion by doing this:
`generated += generateFromCollection(target.getFacts() != null ? target.getFacts().entrySet() : null, STRING_ENTRY);`
and letting the `generateFromCollection` do it's own handling

Reply:
Done.